default.nix

```diff
@@ -1,27 +1,25 @@
 { system ? builtins.currentSystem
 , crossSystem ? null
 , config ? {}
-# Import IOHK common nix lib
-, sourcesOverride ? {}
-# Use pinned Nixpkgs with Haskell.nix overlay
-, pkgs ? import ./nix { inherit system crossSystem config sourcesOverride; }
+# Import pinned Nixpkgs with iohk-nix and Haskell.nix overlays
+, pkgs ? import ./nix/default.nix { inherit system crossSystem config sourcesOverride; }
 # Use this git revision for stamping executables
 , gitrev ? pkgs.commonLib.commitIdFromGitRepoOrZero ./.git
+# Use this to reference local sources rather than the pinned versions (see nix/default.nix)
+, sourcesOverride ? {}
```

Comment:
How do developers use this parameter for their builds? What is the format of the attrset?

Reply:
The format is the same as the one returned from the `sources.nix`. This is documented nowhere it seems, so I'll attempt to do so here.

The `sourcesOverride` should be an attrset with names being one of the source names in `nix/sources.json`. In cardano-wallet these are at the moment: `haskell.nix` `iohk-nix` `niv` `nixpkgs-19.03`.
The value then can be a simple path to a local checkout of those repos, for example if you want to quickly experiment with some changes without having to make commits and updating the sources.json all the time.

Example usage:

    nix build -f . stackNixRegenerate --arg sourcesOverride '{ nixpkgs = ~/github/nixos/nixpkgs; iohk-nix = ~/github/input-output-hk/iohk-nix; }'

The values don't have to be paths, they can also be simply the result of a fetchTarball, fetchGit, fetchurl, fetchFromGitHub, or whatever else produces paths once evaluated.

Reply:
The documentation has not been copied over indeed, though most other project do document the parameter, including the skeleton project:
https://github.com/input-output-hk/iohk-nix/blob/master/skeleton/default.nix#L12-L18

Reply:
Thanks for the info @manveru @jbgi. I have added comments accordingly.